### DISCUSSION TITLE
Zephyr and TLS

### DISCUSSION THREAD

**Discussion by xhpohanka** [cesanta/mongoose#2628] [Ask questions, get answers] [answered]
Hello,
I was shortly playing with zephyr device dashboard example on my stm32h745 board but I was not successful in enabling TLS. Is it supported? I have tried `-DMG_TLS=MG_TLS_MBED` and `-DMG_TLS=MG_TLS_BUILTIN` but the site is not loading in both cases?

Can you give me a hint how to continue? Zephyr has own implementation of TLS sockets, can it be used with mongoose?

best regards
Jan

**Comment by scaprile** [comment, marked as answer]
All the other Zephyr examples are built to use MbedTLS. Did you see any of them ?
https://github.com/cesanta/mongoose/blob/813489dc2fa5fe6c718e9e04dc1204b0c3ad32cc/examples/zephyr/http-server/CMakeLists.txt#L9
https://github.com/cesanta/mongoose/blob/813489dc2fa5fe6c718e9e04dc1204b0c3ad32cc/examples/zephyr/http-server/prj.conf#L19-L23
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation/),  [examples](https://github.com/cesanta/mongoose/tree/master/examples/) and [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples) available.

> the site is not loading

Mongoose has logs, Zephyr has logs, and you can sniff your network with Wireshark. Are you at least able to serve a site with no TLS ?
If there is an actual issue, please open an issue, _honor the issue template_ so we can know your scenario and reproduce the issue if necessary, and let us know _exactly_ what you are doing
